### PR TITLE
Dev staging: bringing working changes into dev

### PR DESCRIPTION
**Pull request by amrosnik**
This staging branch handles merge conflicts between `dev` and `working` in a safe way.

The main difference here is that now the way to run the `run_X.py` scripts involves an input file full of input args rather than command line args. 

There's also some revision of the refactors made to `run_southafrica.py`. 

Let's hope this works!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,6 +32,23 @@ to tokenize the text and search for *actual* instances of those keywords.
...
+
+Python packages (these all can be installed via `pip`, and probably other alternatives): 
+* Selenium
+* argparse
+* numpy
+* pandas
+* spacy
+* pytesseract
+* PIL 
+* wand.image
```
Comment (mathematicalmichael) [inline README.md]: argparse and PIL are part of the std library.

in `setup.cfg` I defined the following requirements by guess/checking my way to getting things to run

```
install_requires =
                numpy
                pandas
                spacy
                pdfminer.six
                pytesseract 
                wand 
                selenium
                chromedriver
                webdriver-manager
```

Comment (mathematicalmichael) [reply README.md]: `pip install (-e) .` will collect and install all of these.
I would mention something about the spacy extension instructions / that it auto-downloads if missing.

File: src/legiscrapor/legisweb_class.py [hunk 2]
```diff
@@ -54,142 +100,174 @@ def checkers(self):
       print("You have initialized an automated search of ",self.country," 's legislation and will download files to ",self.downloadPath)
   
   def teardown(self):
-    ## how we quit Chromedriver when we are finished 
-    self.driver.quit()
+      ## how we quit Chromedriver when we are finished
```
Comment (mathematicalmichael) [inline src/legiscrapor/legisweb_class.py:103]: thank you for handling the linting here. I was planning on it at some point.

File: tests_to_write.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
+# Tests to write
+
+This is a set of notes on tests that need to get written to ensure this repo / package 
+actually can follow a test-driven development framework.
```
Comment (mathematicalmichael) [inline tests_to_write.md:4]: <3 definitely.

In `setup.cfg` test-related requirements can be added to separate them from the runtime requirements with `pip install .[test]` grabbing everything needed for local development. I threw some basic ones in there for the `pytest` framework (and iirc, code-coverage). would love to contribute github-actions workflows to automate testing as part of PRs (CI, eventually CD).

File: README.md (hunk 1 above)
Comment (amrosnik) [reply README.md]: cool, on it like a bonnet. Will be making a commit with these changes shortly.

File: tests_to_write.md (hunk 3 above)
Comment (amrosnik) [reply tests_to_write.md:4]: Love it, VERY excited to get this working and would love to work with you to get CI going!

**Pull request merged**